File: GO/Aves.go
package main

// Interface para los métodos abstractos
type Aves interface {
	PonerHuevo() string
}

// Struct “base” que simula la clase padre
type BaseAves struct {
	Nombre string
	Color string
	Pico string
}

// Constructor para BaseAves
func NewBaseAves(nombre, color, pico string) *BaseAves {
	return &BaseAves{
		Nombre: nombre,
		Color: color,
		Pico: pico,
	}
}

File: C#/C#/clases_abstractas/Aves.cs
using System;

namespace C_.clases_abstractas
{
    internal abstract class Aves
    {
        public String nombre;
        public String color;
        public String pico;

        // Constructor con parametros
        public Aves(String nombre, String color, String pico)
        {
            this.nombre = nombre;
            this.color = color;
            this.pico = pico;
        }

        // Metedo abstracto
        public abstract void ponerHuevo();
    }
}
